lib/src/components/segmented_control/segmented_control.dart

```diff
@@ -0,0 +1,632 @@
+      ..add(
+        ObjectFlagProperty<void Function(T p1)?>.has('onChanged', onChanged),
+      )
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(IterableProperty<ZetaButtonSegment<T>>('segments', segments))
+      ..add(DiagnosticsProperty<T>('selected', selected));
+  }
+}
+
+class _ZetaSegmentedControlState<T> extends State<ZetaSegmentedControl<T>>
+    with TickerProviderStateMixin<ZetaSegmentedControl<T>> {
+  T? _highlighted;
+  final _tap = TapGestureRecognizer();
+  Animatable<Rect?>? _thumbAnimatable;
+  late final AnimationController _thumbController = AnimationController(
+    duration: kThemeAnimationDuration,
+    value: 0,
+    vsync: this,
+  );
+
+  late final _thumbScaleAnimation = _thumbScaleController.drive(Tween<double>(begin: 1));
+
+  late final _thumbScaleController = AnimationController(
+    duration: kThemeAnimationDuration,
+    value: 0,
+    vsync: this,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+
+    _tap.onTapUp = _onTap;
+    _highlighted = widget.selected;
+  }
+
+  void _onTap(TapUpDetails details) {
+    final segment = _segmentForXPosition(details.localPosition.dx);
+    if (segment != widget.selected) widget.onChanged?.call(segment);
+  }
+
+  T _segmentForXPosition(double dx) {
+    final RenderBox renderBox = context.findRenderObject()! as RenderBox;
+    final int numOfChildren = widget.segments.length;
+
+    final int index =
+        (dx ~/ (renderBox.size.width / numOfChildren)).clamp(0, numOfChildren - 1); // ignore_clamp_double_lint
+
+    final values = widget.segments.map((e) => e.value).toList();
+    return values.elementAt(index);
+  }
+
+  @override
+  void didUpdateWidget(ZetaSegmentedControl<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (_highlighted != widget.selected) {
+      _thumbController.animateWith(
+        SpringSimulation(
+          const SpringDescription(mass: 1, stiffness: 500, damping: 44),
+          0,
+          1,
+          0, // Every time a new spring animation starts the previous animation stops.
+        ),
+      );
+      _thumbAnimatable = null;
+      _highlighted = widget.selected;
+    }
+  }
+
+  @override
+  void dispose() {
+    _thumbScaleController.dispose();
+    _thumbController.dispose();
+
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = [];
+    int index = 0;
+    int? highlightedIndex;
+    for (final segment in widget.segments) {
+      final isHighlighted = _highlighted == segment.value;
+      if (isHighlighted) highlightedIndex = index;
+
+      if (index != 0) {
+        children.add(SizedBox(key: Key(index.toString())));
+      }
+
+      children.add(
+        _Segment<T>(
+          key: ValueKey<T>(segment.value),
+          rounded: widget.rounded,
+          child: segment.child,
+        ),
+      );
+
+      index += 1;
+    }
+
+    final colors = Zeta.of(context).colors;
+
+    return Container(
```

Comment:
could we wrap this container in 2 widgets:
* `MouseRegion` - so that we can use the click cursor when the component is used on non-mobile platforms
* `SelectionContainer.disabled` - so that text can not be selected